modules/kafka/kafka-dest-driver.c

```diff
@@ -163,7 +197,8 @@ _kafka_delivery_report_cb(rd_kafka_t *rk,
       LogPathOptions path_options = LOG_PATH_OPTIONS_INIT;
 
       msg_debug("kafka: delivery report for message came back with an error, putting it back to our queue",
-                evt_tag_str("topic", self->topic_name),
+                evt_tag_str("topic", self->topic_name->template),
```

Comment:
Can you explain me the debug_flag part?

Reply:
I have a small concern. In case we evaluate the template inside the delivery report call back function, can the latency affect the topic name displayed here vs the actual topic where the message was published to?

Reply:
Most of the cases it won't be a problem, because the resolution is only defined by the message. If however there is a template function, which uses something else besides the message (like random generation, or system time, etc...) the resolution will not be the same, indeed.

Reply:
Although I agree it would be useful, reformatting it again does not feel right in my opinion. If we wanted to include per-message information like this we would start thinking about using an msg_opaque that is not pointing directly to LogMessage, rather a structure that encapsulates both the LogMessage and any information we might need in the delivery callback (e.g. topic name).

I don't think the topic name would include anything that is not part of the formatted log message. So I would wait until this becomes an issue :)